apps/src/code-studio/components/progress/LessonLockDialog.jsx

```diff
@@ -11,264 +10,287 @@ import SectionSelector from './SectionSelector';
 import {teacherDashboardUrl} from '@cdo/apps/templates/teacherDashboard/urlHelpers';
 import {NO_SECTION} from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
 
-class LessonLockDialog extends React.Component {
-  static propTypes = {
-    isOpen: PropTypes.bool.isRequired,
-    handleClose: PropTypes.func.isRequired,
-    initialLockStatus: PropTypes.arrayOf(
-      PropTypes.shape({
-        name: PropTypes.string.isRequired,
-        lockStatus: PropTypes.oneOf(Object.values(LockStatus)).isRequired
-      })
-    ),
-    selectedSectionId: PropTypes.number,
-    saving: PropTypes.bool.isRequired,
-    saveDialog: PropTypes.func.isRequired
-  };
-
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      lockStatus: props.initialLockStatus
-    };
-  }
+function LessonLockDialog({
+  isOpen,
+  handleClose,
+  initialLockStatus,
+  selectedSectionId,
+  saving,
+  saveDialog
+}) {
+  const [lockStatuses, setlockStatuses] = useState(initialLockStatus);
```

Comment:
nit: `setLockStatuses`

Reply:
nit: since `initialLockStatus` is an array and being set to `lockStatuses`, it may make sense to rename it to `initialLockStatuses`

Reply:
@bethanyaconnor, good catch on `setLockStatuses`, thanks!

@maureensturgeon, haha, I actually had that change in at some point.  I ended up leaving it as `initialLockStatus` because I wanted it to be obvious that the redux integration here didn't change at all and I'm expecting to remove this property entirely in the next PR.